examples/differential_correction_coupled.py

```diff
@@ -0,0 +1,223 @@
+# -*- coding: utf-8 -*-
+"""
+Created on Thu Oct 25 17:19:30 2019
+
+@author: Wenyang Lyu and Shibabrat Naik
+
+Script to compute unstable periodic orbits at specified energies for the 
+coupled quartic Hamiltonian using differential correction
+"""
+
+import numpy as np
+from scipy.integrate import solve_ivp
+import time
+
+import matplotlib as mpl
+from matplotlib import cm
+from mpl_toolkits.mplot3d import Axes3D
+import matplotlib.pyplot as plt
+mpl.rcParams['mathtext.fontset'] = 'cm'
+mpl.rcParams['mathtext.rm'] = 'serif'
+
+# This needs testing for installation 
+import sys
+sys.path.insert(0, './src/')
+import differential_correction as diffcorr
+import coupled_quartic_hamiltonian as coupled
+# This needs testing for installation 
+
+#%% Setting up parameters for the method and the Hamiltonian system
+
+save_final_plot = True
+show_final_plot = False
+N = 4               # dimension of phase space
+omega=1.00
+EPSILON_S = 0.0     # Energy of the saddle
+alpha = 1.00
+beta = 1.00
+epsilon= 1e-1
+parameters = np.array([1, omega, EPSILON_S, alpha, beta, omega, epsilon])
+eqNum = 1
+#deltaE_vals = [0.01, 0.1, 1.00, 2.0, 4.0]
+#linecolor = ['b','r','g','m','c']
+deltaE_vals = [0.1, 1.00]
+linecolor = ['b','r']
+
+
+eqPt = diffcorr.get_eq_pts(eqNum, coupled.init_guess_eqpt_coupled, \
+                            coupled.grad_pot_coupled, parameters)
+
+#energy of the saddle eq pt
+eSaddle = diffcorr.get_total_energy([eqPt[0],eqPt[1],0,0], \
+                                    coupled.pot_energy_coupled, parameters) 
+#If orbit is an n-array, e.g. orbit = [orbit_0, orbit_1, ..., orbit_n]
+# 
+#e = np.zeros(n,1)    
+#for i in range(n):
+#    e[i] = get_total_energy_deleonberne(orbit[i], parameters)
+#e = np.mean(e)
+
+#%%
+nFam = 100 # use nFam = 10 for low energy
+
+# first two amplitudes for continuation procedure to get p.o. family
+Ax1  = 2.e-5 # initial amplitude (1 of 2) values to use: 2.e-3
+Ax2  = 2*Ax1 # initial amplitude (2 of 2)
+
+t = time.time()
+
+#  get the initial conditions and periods for a family of periodic orbits
+with open("x0_diffcorr_fam_eqPt%s_coupled.dat" %eqNum,'a+') as po_fam_file:
+    [po_x0Fam,po_tpFam] = diffcorr.get_po_fam(eqNum, Ax1, Ax2, nFam, \
+                                            po_fam_file, \
+                                            coupled.init_guess_eqpt_coupled, \
+                                            coupled.grad_pot_coupled, \
```

Comment:
`coupled` should not be appearing twice here.

Maybe each example should be structured as a class that contains the relevant functions.

Reply:
We are passing two functions that do different things for the given example system, so I think this explicit mention keeps the code readable. 

On structuring the example as a class issue, I think abstracting the examples as classes will lead to the package being obscure for a beginner. The example system, in our vision for this package, is like a new problem that a user would like to solve using one of the methods so converting them into a class will force them to implement the new problem as a class. These examples aren't meant to be general systems so I don't see abstraction being useful for the examples.

Reply:
@dpsanders @Shibabrat This seems like it may be a good future development but not necessary at this stage.

@dpsanders please let me know if you are willing to agree.

Reply:
Yes OK. I'm just trying to reduce duplication and make it clearer what the structure is.

Reply:
@dpsanders Thank you for your excellent code review. Let us proceed with the JOSS submission and @Shibabrat can pursue this improvement in a future iteration of the package.